modules/aws-eks-teams/locals.tf

```diff
@@ -0,0 +1,33 @@
+locals {
+  team_manifests = flatten([
+    for team_name, team_data in var.application_teams :
+    fileset(path.root, "${team_data.manifests_dir}/*")
+  ])
+
+  eks_oidc_issuer_url   = replace(data.aws_eks_cluster.eks_cluster.identity[0].oidc[0].issuer, "https://", "")
+  eks_oidc_provider_arn = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${local.eks_oidc_issuer_url}"
```

Comment:
nit but the formatted string here could leverage your `partition` and `account_id` locals as well?